lib/capistrano/tasks/assets.rake

```diff
@@ -75,15 +76,13 @@ namespace :deploy do
     task :restore_manifest do
       on release_roles(fetch(:assets_roles)) do
         within release_path do
-          source = release_path.join('assets_manifest_backup')
-          target = capture(:ls, release_path.join('public', fetch(:assets_prefix),
-                                                  'manifest*')).strip
-          if test "[[ -f #{source} && -f #{target} ]]"
-            execute :cp, source, target
-          else
-            msg = 'Rails assets manifest file (or backup file) not found.'
-            warn msg
-            fail Capistrano::FileNotFound, msg
+          source = release_path.join('tmp', 'capistrano_assets_backup', 'manifest*')
+          target = release_path.join('public', fetch(:assets_prefix))
+          manifests = capture(:ls, release_path.join('public', fetch(:assets_prefix), 'manifest*')).split
+
+          execute :cp, source, target
+          manifests.each do |manifest|
+            info "#{manifest.split('/').last} restored."
```

Comment:
`File.basename` or `Pathname#basename` should be used instead of `split('/').last`